kafka-0.10.x/src/main/scala/monix/kafka/KafkaConsumerObservable.scala

```diff
@@ -100,6 +107,27 @@ trait KafkaConsumerObservable[K, V, Out] extends Observable[Out] {
     // at most once
     cancelTask.memoize
   }
+
+  /* Returns task that constantly polls the `KafkaConsumer` in case subscriber
+   * is still processing last fed batch.
+   * This allows producer process commit calls and also keeps consumer alive even
+   * with long batch processing.
+   */
+  private def pollConsumer(consumer: KafkaConsumer[K, V]): Task[Unit] = {
+    Task
+      .sleep(config.pollInterval)
+      .flatMap { _ =>
+        if (!isAcked) {
+          Task.evalAsync {
+            consumer.synchronized {
+              blocking(consumer.poll(0))
```

Comment:
@Avasil @voidconductor Looking at the alpakka implementation, they raise an error here if the poll result size != 0 (we might lose messages if its > 0). Should we do the same thing? Did you test that it won't overlap with the poll during `ackTask`?

Reply:
All polls are synchronized on consumer, therefore they can't overlap.

About probability of losing records - consumer is paused right after `poll`, according to  `pause` javadoc:

```
     * Suspend fetching from the requested partitions. Future calls to {@link #poll(long)} will not return
     * any records from these partitions until they have been resumed using {@link #resume(TopicPartition...)}.
```

Here we call `poll(0)` without resuming, so it shouldn't return any records, and if it does - it will be apache kafka bug

Reply:
It seems like many implementations do this check (though they don't use `synchronized` I think), I will try to double check if it concerns us or not in the following days

Reply:
I've looked around a bit, it seems like it is possible that `poll(0)` with paused partitions could receive new partition assignment with new records for it

I haven't reproduced it yet, I will have to setup something because unfortunately I don't have any codebase to play around right now :D 

If you'd like to experiment, I published a snapshot from this branch: 
`1.0.0-RC5-d7546d6-SNAPSHOT`

Few links from other libraries:
- https://gitter.im/ovotech/fs2-kafka?at=5d67f3f506e77851a3122b97
- https://github.com/zio/zio-kafka/pull/40

Reply:
I've found out we can use `commitAsync(Map.empty, null)` instead of `poll(0)` to complete commit callbacks, as it also calls coordinator's `invokeCompletedOffsetCommitCallbacks` internally

This means no risk of getting unexpected messages or losing data

There are downsides I see in this approach:
- looks like metadata is not requested and rebalance is not performed before next poll
- it won't refresh `max.poll.interval` timeout, so https://github.com/monix/monix-kafka/issues/101 will not be solved

Reply:
>  as it also calls coordinator's invokeCompletedOffsetCommitCallbacks internally

I'm also worried about depending on implementation detail (unless docs somewhere guarantee it) that could perhaps silently change between versions